GW2EIEvtcParser/EncounterLogic/Strikes/EndOfDragons/AetherbladeHideout.cs

```diff
@@ -268,5 +277,87 @@ internal override FightData.EncounterMode GetEncounterMode(CombatData combatData
+            
+            if(log.FightData.Success)
+            {
+                if (triangulation.Any())
+                {
+                    foreach (Player p in log.PlayerList)
+                    {
+                        if (p.HasBuff(log, AchievementEligibilityTriangulation, log.FightData.FightEnd - ServerDelayConstant))
+                        {
+                            InstanceBuffs.Add((log.Buffs.BuffsByIds[AchievementEligibilityTriangulation], 1));
+                            hasTriangulationBeenAdded = true;
+                            break;
+                        }
+                    }
+                }
+                if (!hasTriangulationBeenAdded && CustomCheckTriangulationEligibility(log))
+                {
+                    InstanceBuffs.Add((log.Buffs.BuffsByIds[AchievementEligibilityTriangulation], 1));
+                }
+            }
+        }
+
+        private static bool CustomCheckTriangulationEligibility(ParsedEvtcLog log)
+        {
+            IReadOnlyList<long> beamsBuffs = new List<long>() { MaiTrinCMBeamsTargetBlue, MaiTrinCMBeamsTargetGreen, MaiTrinCMBeamsTargetRed };
+            var beamsSegments = new List<Segment>();
+            var bombInvulnSegments = new List<Segment>();
+
+            foreach (AgentItem player in log.PlayerAgents)
+            {
+                IReadOnlyDictionary<long, BuffsGraphModel> bgms = log.FindActor(player).GetBuffGraphs(log);
+                foreach (long buff in beamsBuffs)
+                {
+                    beamsSegments = GetBuffSegments(bgms, buff, beamsSegments).OrderBy(x => x.Start).ToList();
+                }
+            }
+
+            foreach (AgentItem agent in log.AgentData.GetNPCsByID(ArcDPSEnums.TrashID.FerrousBomb))
+            {
+                IReadOnlyDictionary<long, BuffsGraphModel> bgms = log.FindActor(agent).GetBuffGraphs(log);
+                bombInvulnSegments = GetBuffSegments(bgms, FailSafeActivated, bombInvulnSegments).OrderBy(x => x.Start).ToList();
+            }
+
+            int counter = 0;
+
+            // For each segment where a bomb is invulnerable, check if it has started between the assignment and loss of a beam effect on a player
+            // If the counter is == 8, it means every possible combination check has been met, meaning it's eligible for the achievement.
+            foreach (Segment invuln in bombInvulnSegments)
+            {
+                foreach (Segment s in  beamsSegments)
+                {
+                    if (s.Start < invuln.Start && invuln.Start < s.End)
+                    {
+                        counter++;
+                    }
+                }
+            }
+
+            return counter == 8;
+        }
+
+        private static List<Segment> GetBuffSegments(IReadOnlyDictionary<long, BuffsGraphModel> bgms, long buff, List<Segment> segments)
+        {
+            if (bgms != null && bgms.ContainsKey(buff))
```

Comment:
ContainsKey(buff) to be replaced by TryGetValue(buff, out BuffsGraphModel bgm), remove the one below